authapi/permissions.py

```diff
@@ -162,3 +172,52 @@ def object_permission_set(self):
                 )
             )
         )
+
+
+class UserPermission(BaseComposedPermision):
+    '''Permissions for the UserViewSet.'''
+    def global_permission_set(self):
+        '''All users must be authenticated. Only admins can create other admin
+        users.'''
+        only_admins_create_admins = Or(
+            AllowAdmin,
+            And(
+                ObjAttrTrue(
+                    lambda r, _: r.data.get('admin').lower() != 'true'),
+                Or(
+                    AllowPermission('user:create'),
+                    AllowPermission('org:admin')
+                )
+            )
+        )
+
+        return And(
+            AllowOnlyAuthenticated,
+            Or(
+                Not(AllowCreate),
+                only_admins_create_admins
+            )
+        )
+
+    def object_permission_set(self):
+        '''All users have view permissions. Admin users, and users with
+        org:admin can create, update, and delete any user. Any user can update
+        or delete themselves. Users with user:create permission can create
+        new users. Only admins can create or modify other admin users.'''
+        return Or(
+            AllowOnlySafeHttpMethod,
+            AllowAdmin,
+            And(
+                AllowPermission('org:admin'),
+                ObjAttrTrue(lambda _, u: not u.is_superuser),
```

Comment:
`u` here is not the user that is making the request, it's the object that they're trying to modify. So here we're saying that `org:admin` may only modify the user if `u.is_superuser` is false, ie the user they are trying to modify isn't an admin user.